lib/galaxy/webapps/galaxy/api/biotools.py

```diff
@@ -0,0 +1,299 @@
+    :param tool_id: Galaxy tool id
+    :param tool_version: Galaxy tool version
+    :return: tool registry name
+    :rtype: string
+    """
+    tbl = string.maketrans('.:/', '___')
+    # with a long id galaxy, xml version is already in the id
+    if tool_version in tool_id:
+        return str(tool_id).translate(tbl)
+    else:
+        return str("_".join([tool_id, tool_version])).translate(tbl)
+
+
+def format_description(description):
+    """
+    Test the first and last char of a description and replace them
+    with the format adapted to Elixir
+    :param description: Tool Galaxy description
+    :return: Formatted tool Galaxy description
+    :rtype: string
+    """
+    try:
+        size = len(description)
+        if description[size - 1] == '.':
+            return description[0].upper() + description[1:size]
+        else:
+            return description[0].upper() + description[1:size] + '.'
+    except IndexError:
+        raise description
+
+
+def get_source_registry(tool_id):
+    """
+    Take a tool id and return the toolshed url if the id is a long id or a empty string if id is a short id
+    :param tool_id: Galaxy tool id
+    :return: toolshed url for a tool or empty string
+    :rtype: string
+    """
+    try:
+        return "/".join(tool_id.replace('repos', 'view', 1).split('/')[0:-2])
+    except ValueError:
+        log.warning("Not well formated tool id: {}".format(tool_id))
+        return ""
+
+
+def build_general_dict(tool_meta_data):
+    """
+    Extract informations from a galaxy json tool and return the general json in the biotools format
+    :param tool_meta_data: galaxy json tool
+    :return: biotools dictionary
+    :rtype: dictionary
+    """
+
+    gen_dict = {
+        'version': tool_meta_data[u'tool_version'],
+        'description': format_description(tool_meta_data[u'description']),
+        'uses': [{
+            "usesName": tool_meta_data[u'tool_id'],
+            "usesHomepage": "{0}?tool_id={1}".format(os.path.join('conf.galaxy_url', "root"),
+                                                     tool_meta_data[u'tool_id']),
+            "usesVersion": tool_meta_data[u'tool_version']
+        }],
+        'collection': '',
+        'sourceRegistry': get_source_registry(tool_meta_data[u'tool_id']),
+        'resourceType': ["Tool"],
+        'maturity': 'Stable',
+        'platform': [{u'term': 'Linux'}],
+        'interface': [{
+            'interfaceType': "Web UI",
+            'interfaceDocs': '',
+            'interfaceSpecURL': '',
+            'interfaceSpecFormat': ''
+        }],
+        'topic': [{
+            'uri': "http://edamontology.org/topic_0003",
+            'term': "EDAM label placeholder"
+        }],
+        'publications': {'publicationsPrimaryID': "None", 'publicationsOtherID': []},
+        'homepage': 'conf.galaxy_url',
+        'accessibility': '"Private" if conf.private else "Public"',
```

Comment:
I don't think galaxy knows whether it is a public instance or not. But it's not mandatory, so let's remove this.